src/main/java/org/c02e/jpgpj/Encryptor.java

```diff
@@ -424,13 +482,74 @@ public void encrypt(InputStream plaintext, OutputStream ciphertext,
         SigningOutputStream signingstream = sign(ciphertext, meta);
         ciphertext = pipeline(signingstream, stack);
         ciphertext = pipeline(packet(ciphertext, meta), stack);
+        return new EncryptorWrapperStream(ciphertext, signingstream, stack);
+    }
+
+    protected static class EncryptorWrapperStream extends FilterOutputStream {
+        protected final AtomicBoolean finished = new AtomicBoolean(false);
+        protected final SigningOutputStream signingstream;
+        protected final List<? extends OutputStream> stack;
+        protected final byte[] oneByte = { 0 };
+        protected EncryptorWrapperStream(
+                OutputStream ciphertext, SigningOutputStream signer, List<? extends OutputStream> wrappers) {
+            super(ciphertext);
+
+            signingstream = signer;
+            stack = wrappers;
+        }
+
+        @Override
+        public void write(int b) throws IOException {
+            oneByte[0] = (byte) b;
+            write(oneByte, 0, 1);
+        }
+
+        @Override   // just making sure
+        public void write(byte b[]) throws IOException {
+            write(b, 0, b.length);
+        }
+
+        @Override
+        public void write(byte[] b, int off, int len) throws IOException {
+            // FilterOutputStream implements it by writing one byte at a time
+            if (signingstream != null) {
+                signingstream.update(b, off, len);
+            }
+            out.write(b, off, len);
+        }
+
+        @Override
+        public void close() throws IOException {
+            // Ignore if already closed
+            if (finished.getAndSet(true)) {
+                return;
+            }
+
+            flush();
```

Comment:
will this work okay without calling flush()? -- I would like to avoid flushing the underlying output stream if at all possible (to allow users as much leeway about when to write out the bytes to disk or wire)

Reply:
I'll look into it, but it might be safe to flush since the output stream might actually be wrapped inside several other - some of which might maintain a buffer.

Reply:
After reviewing the code - will I agree with you that we should avoid flushing I thing it is the safer choice in view of the possible buffering of one of the wrappers.